crates/rooch-executor/src/actor/executor.rs

```diff
@@ -125,6 +125,20 @@ impl ExecutorActor {
             .moveos_store
             .handle_tx_output(tx_hash, output.clone())?;
 
+        // The cost table has been upgraded, we need to reload the native functions.
+        if self.moveos.cost_table.read().is_none() {
```

Comment:
there are another moveos instance in ReadExecutor

Reply:
There should be a formal solution to provide a global, asynchronous, and thread-safe subscription and notification mechanism for notifying successful gas upgrades.